modules/tests/test/create_vm_from_manifest_test.go

```diff
@@ -242,6 +249,197 @@ var _ = Describe("Create VM from manifest", func() {
+			config := &testconfigs.CreateVMTestConfig{
+				TaskRunTestConfig: testconfigs.TaskRunTestConfig{
+					ServiceAccount: CreateVMFromManifestServiceAccountName,
+					ExpectedLogs:   ExpectedSuccessfulVMCreation,
+				},
+				TaskData: testconfigs.CreateVMTaskData{
+					Virtctl: fmt.Sprintf("--name %s --instancetype %s", vmName, instancetypeName),
+				},
+			}
+			f.TestSetup(config)
+
+			createInstancetype(f, instancetypeName)
+
+			expectedVMStub := config.TaskData.GetExpectedVMStubMeta()
+			f.ManageVMs(expectedVMStub)
+
+			runner.NewTaskRunRunner(f, config.GetTaskRun()).
+				CreateTaskRun().
+				ExpectSuccess().
+				ExpectLogs(config.GetAllExpectedLogs()...)
+
+			vm, err := vm.WaitForVM(f.KubevirtClient, f.TestOptions.DeployNamespace, vmName,
+				"", config.GetTaskRunTimeout(), false)
+
+			Expect(err).ShouldNot(HaveOccurred())
+
+			Expect(vm.Name).To(Equal(vmName))
+			Expect(*vm.Spec.RunStrategy).To(Equal(kubevirtv1.RunStrategyAlways))
+
+			deleteInstancetype(f, instancetypeName)
+		})
+
+		It("should start with startVM set to true", func() {
+			vmName := "my-vm-3"
+			instancetypeName := "instancetype-3"
+			config := &testconfigs.CreateVMTestConfig{
+				TaskRunTestConfig: testconfigs.TaskRunTestConfig{
+					ServiceAccount: CreateVMFromManifestServiceAccountName,
+					ExpectedLogs:   ExpectedSuccessfulVMCreation,
+				},
+				TaskData: testconfigs.CreateVMTaskData{
+					Virtctl: fmt.Sprintf("--name %s --run-strategy=Halted --instancetype %s", vmName, instancetypeName),
+					StartVM: "true",
+				},
+			}
+			f.TestSetup(config)
+
+			createInstancetype(f, instancetypeName)
+
+			expectedVMStub := config.TaskData.GetExpectedVMStubMeta()
+			f.ManageVMs(expectedVMStub)
+
+			runner.NewTaskRunRunner(f, config.GetTaskRun()).
+				CreateTaskRun().
+				ExpectSuccess().
+				ExpectLogs(config.GetAllExpectedLogs()...)
+
+			vm, err := vm.WaitForVM(f.KubevirtClient, f.TestOptions.DeployNamespace, vmName,
+				"", config.GetTaskRunTimeout(), false)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			Expect(vm.Name).To(Equal(vmName))
+			// startVM should set RunStrategy to Always
+			Expect(*vm.Spec.RunStrategy).To(Equal(kubevirtv1.RunStrategyAlways))
+
+			deleteInstancetype(f, instancetypeName)
+		})
+
+		It("should fail with invalid params", func() {
+			config := &testconfigs.CreateVMTestConfig{
+				TaskRunTestConfig: testconfigs.TaskRunTestConfig{
+					ServiceAccount: CreateVMFromManifestServiceAccountName,
+					ExpectedLogs:   "unknown flag: --invalid",
+				},
+				TaskData: testconfigs.CreateVMTaskData{
+					Virtctl: "--invalid params",
+				},
+			}
+			f.TestSetup(config)
+
+			expectedVMStub := config.TaskData.GetExpectedVMStubMeta()
+			f.ManageVMs(expectedVMStub)
+
+			runner.NewTaskRunRunner(f, config.GetTaskRun()).
+				CreateTaskRun().
+				ExpectFailure().
+				ExpectLogs(config.GetAllExpectedLogs()...)
+		})
+
+		DescribeTable("should fail creating VM", func(config *testconfigs.CreateVMTestConfig) {
```

Comment:
Isn't this more of a general test case that affects every creation mode? I would have placed it on a higher level.